src/main/java/fr/openent/competences/model/ParamsBulletins.java

```diff
@@ -0,0 +1,225 @@
+                .put(BILAN_PER_DOMAINE_LIBELLE, getLibelle("evaluations.bilan.by.domaine"))
+                .put(LEVEL_STUDENT, getLibelle("level.student"))
+                .put(LEVEL_CLASS, getLibelle("level.class"))
+                .put(AVERAGE_STUDENT, getLibelle("average.student"))
+                .put(AVERAGE_CLASS, getLibelle("average.class"))
+                .put(LEVEL_ITEMS, getLibelle("level.items"))
+                .put(AVERAGES, getLibelle("average"))
+                .put(EVALUATION_BY_DOMAINE, getLibelle("evaluation.by.domaine"))
+                .put(COEFFICIENT_LIBELLE, getLibelle("viescolaire.utils.coef"))
+                .put(MOYENNE_ANUELLE_LIBELLE, getLibelle("average.annual"))
+                .put(MOYENNE_GENERALE_LIBELLE, getLibelle("average.general"));
+    }
+
+    public void setParams(JsonObject otherParams) {
+        params.put(GET_RESPONSABLE, otherParams.getBoolean(GET_RESPONSABLE))
+                .put(GET_MOYENNE_CLASSE, otherParams.getBoolean(MOYENNE_CLASSE))
+                .put(GET_MOYENNE_ELEVE, otherParams.getBoolean(MOYENNE_ELEVE))
+                .put(GET_STUDENT_RANK, otherParams.getBoolean(STUDENT_RANK))
+                .put(GET_CLASS_AVERAGE_MINMAX, otherParams.getBoolean(CLASS_AVERAGE_MINMAX))
+                .put(GET_POSITIONNEMENT, otherParams.getBoolean(POSITIONNEMENT))
+                .put(SHOW_PROJECTS, otherParams.getBoolean(SHOW_PROJECTS))
+                .put(SHOW_FAMILY, otherParams.getBoolean(SHOW_FAMILY))
+                .put(GET_PROGRAM_ELEMENT, otherParams.getBoolean(GET_PROGRAM_ELEMENT))
+                .put(SHOW_BILAN_PER_DOMAINE, otherParams.getBoolean(SHOW_BILAN_PER_DOMAINE))
+                .put(IMG_STRUCTURE, otherParams.getString(IMG_STRUCTURE))
+                .put(HAS_IMG_STRUCTURE, otherParams.getBoolean(HAS_IMG_STRUCTURE))
+                .put(IMG_SIGNATURE, otherParams.getString(IMG_SIGNATURE))
+                .put(HAS_IMG_SIGNATURE, otherParams.getBoolean(HAS_IMG_SIGNATURE))
+                .put(NAME_CE, otherParams.getString(NAME_CE))
+                .put(PRINT_COEFFICIENT, otherParams.getBoolean(COEFFICIENT))
+                .put(PRINT_SOUS_MATIERES, otherParams.getBoolean(PRINT_SOUS_MATIERES))
+                .put(PRINT_MOYENNE_ANNUELLE, otherParams.getBoolean(MOYENNE_ANNUELLE))
+                .put(NEUTRE, otherParams.getBoolean(NEUTRE, false))
+                .put(HIDE_HEADTEACHER, otherParams.getBoolean(HIDE_HEADTEACHER,false))
+                .put(ADD_OTHER_TEACHER, otherParams.getBoolean(ADD_OTHER_TEACHER,false))
+                .put(FUNCTION_OTHER_TEACHER, otherParams.getString(FUNCTION_OTHER_TEACHER,""))
+                .put(OTHER_TEACHER_NAME, otherParams.getString(OTHER_TEACHER_NAME,""))
+                .put(AGRICULTURE_LOGO, otherParams.getBoolean(AGRICULTURE_LOGO,false));
+        JsonArray niveauCompetences;
+        try {
+            niveauCompetences = (JsonArray) params.getValue(NIVEAU_COMPETENCE);
```

Comment:
params.getJsonArray ? Cela trown aussi un ClassCastException si c'est pas un JsonArray
 
`  /**
   * Get the JsonArray value with the specified key
   *
   * @param key  the key to return the value for
   * @return the value or null if no value for that key
   * @throws java.lang.ClassCastException if the value is not a JsonArray
   */
  public JsonArray getJsonArray(String key) {
    Objects.requireNonNull(key);
    Object val = map.get(key);
    if (val instanceof List) {
      val = new JsonArray((List)val);
    }
    return (JsonArray)val;
  }`